Makefile

```diff
@@ -408,40 +407,61 @@ static-checks: vendor
 install:
 	CGO_ENABLED=0 go install github.com/projectcalico/calico-containers/calicoctl
 
-## Build a binary for a release
-release-calicoctl: clean update-tools dist/calicoctl test-containerized
-	docker tag calico/calicoctl:$(CALICOCONTAINERS_VERSION) quay.io/calico/calicoctl:$(CALICOCONTAINERS_VERSION)
-	@echo Now attach the binaries to github dist/calicoctl
-	@echo And push the images to Docker Hub and quay.io:
-	@echo docker push calico/calicoctl:$(CALICOCONTAINERS_VERSION)
-	@echo docker push quay.io/calico/calicoctl:$(CALICOCONTAINERS_VERSION)
+release: clean
+ifndef VERSION
+	$(error VERSION is undefined - run using make release VERSION=vX.Y.Z)
+endif
+	git tag $(VERSION)
+
+	$(MAKE) dist/calicoctl calico/ctl calico/node
+	# Check that the version output includes the version specified.
+# Tests that the "git tag" makes it into the binary. Main point is to catch "-dirty" builds
+	if ! dist/calicoctl version | grep -P 'Version:\s*$(VERSION)$$'; then echo "Reported version:" `dist/calicoctl version` "\nExpected version: $(VERSION)"; false; else echo "Version check passed\n"; fi
```

Comment:
It would be nice to run this in the calico/ctl container - that way this should all just work in a OS X environment.